docker/scripts/Test-AzureStack.ps1

```diff
@@ -1,75 +1,309 @@
+{
+    Log -message "Running Get-AzsNetworkQuota..."
+    Get-AzsNetworkQuota | Format-Table -AutoSize | Out-String | Log
+}
 
-    Log -param "END: $($MyInvocation.MyCommand)"
+function AzsStorageAdmin
+{
+    Log -message "Running Get-AzsStorageQuota..."
+    Get-AzsStorageQuota | Format-Table -AutoSize | Out-String | Log
 }
 
-function Test-UserModules()
+function AzsSubscription
 {
-  Log -param "START: $($MyInvocation.MyCommand)"
+    Log -message "Running Get-AzsSubscription..."
+    Get-AzsSubscription | Format-Table -AutoSize | Out-String | Log
+}
 
-  Import-Module Az.Resources
-  Get-AzResourceGroup | Format-Table -AutoSize
+function AzsSubscriptionAdmin
+{
+    Log -message "Running Get-AzsUserSubscription..."
+    Get-AzsUserSubscription | Format-Table -AutoSize | Out-String | Log
+}
 
-  Log -param "END: $($MyInvocation.MyCommand)"
+function AzsUpdateAdmin
+{
+    Log -message "Running Get-AzsUpdate..."
+    Get-AzsUpdate | Format-Table -AutoSize | Out-String | Log
 }
 
-function Main()
+function Main
 {
-    $ErrorActionPreference = 'Stop'
-    Setup
-    Test-AdminModules
-    Test-UserModules
+    Import-Module Az.Accounts
+    Import-Module Az.Resources
+    Import-Module AzureStack
+
+    Get-AzsUpdate | Format-Table -AutoSize | Out-String | Log
+
+    Test -testFunctionName "AzAccounts" -testFunction $Function:AzAccounts
+    Test -testFunctionName "AzBilling" -testFunction $Function:AzBilling
+    Test -testFunctionName "AzCompute" -testFunction $Function:AzCompute
+    Test -testFunctionName "AzKeyVault" -testFunction $Function:AzKeyVault
+    Test -testFunctionName "AzMonitor" -testFunction $Function:AzMonitor
+    Test -testFunctionName "AzResources" -testFunction $Function:AzResources
+    Test -testFunctionName "AzWebsites" -testFunction $Function:AzWebsites
+    Test -testFunctionName "AzsAzureBridgeAdmin" -testFunction $Function:AzsAzureBridgeAdmin
+    Test -testFunctionName "AzsBackupAdmin" -testFunction $Function:AzsBackupAdmin
+    Test -testFunctionName "AzsCommerceAdmin" -testFunction $Function:AzsCommerceAdmin
+    Test -testFunctionName "AzsComputeAdmin" -testFunction $Function:AzsComputeAdmin
+    Test -testFunctionName "AzsFabricAdmin" -testFunction $Function:AzsFabricAdmin
+    Test -testFunctionName "AzsInfrastructureInsightsAdmin" -testFunction $Function:AzsInfrastructureInsightsAdmin
+    Test -testFunctionName "AzsGalleryAdmin" -testFunction $Function:AzsGalleryAdmin
+    Test -testFunctionName "AzsKeyVaultAdmin" -testFunction $Function:AzsKeyVaultAdmin
+    Test -testFunctionName "AzsNetworkAdmin" -testFunction $Function:AzsNetworkAdmin
+    Test -testFunctionName "AzsStorageAdmin" -testFunction $Function:AzsStorageAdmin
+    Test -testFunctionName "AzsSubscription" -testFunction $Function:AzsSubscription
+    Test -testFunctionName "AzsSubscriptionAdmin" -testFunction $Function:AzsSubscriptionAdmin
+    Test -testFunctionName "AzsUpdateAdmin" -testFunction $Function:AzsUpdateAdmin
 }
 
 Main
```

Comment:
I would prefer printing out few version info part of the logs like loaded module version, image versions etc..

Reply:
I don't think we can print out image information. I couldn't find information on getting image information from inside the container.